File: frontend/src/types/external-platform.ts
export interface GoogleReviewSnapshot {
  id: string;
  tenant_id: string;
  location_id: string | null;
  snapshot_date: string;
  rating_average: string | null;
  review_count_total: number;
  new_reviews_count: number;
  positive_count: number;
  neutral_count: number;
  negative_count: number;
  google_place_id: string | null;
  created_at: string;
}

export interface GoogleAdsCampaign {
  id: string;
  tenant_id: string;
  location_id: string | null;
  google_campaign_id: string;
  google_customer_id: string;
  name: string;
  status: string;
  campaign_type: string | null;
}

export interface GoogleAdsDailyMetric {
  id: string;
  campaign_id: string;
  metric_date: string;
  spend: string | null;
  impressions: number;
  clicks: number;
  conversions: string | null;
  roas: string | null;
  currency_code: string;
}

export interface GoogleAdsSummary {
  period_start: string;
  period_end: string;
  total_spend: string;
  total_impressions: number;
  total_clicks: number;
  total_conversions: string;
  average_roas: string | null;
  campaigns: GoogleAdsCampaign[];
  daily_metrics: GoogleAdsDailyMetric[];
}

File: backend/app/schemas/external_platform.py
"""Schemas for external platform data (Google Reviews, Google Ads)."""
import uuid
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, ConfigDict


class GoogleReviewSnapshotResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    location_id: Optional[uuid.UUID] = None
    snapshot_date: str
    rating_average: Optional[Decimal] = None
    review_count_total: int
    new_reviews_count: int
    positive_count: int
    neutral_count: int
    negative_count: int
    google_place_id: Optional[str] = None
    created_at: str


class GoogleAdsCampaignResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    location_id: Optional[uuid.UUID] = None
    google_campaign_id: str
    google_customer_id: str
    name: str
    status: str
    campaign_type: Optional[str] = None


class GoogleAdsDailyMetricResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    campaign_id: uuid.UUID
    metric_date: str
    spend: Optional[Decimal] = None
    impressions: int
    clicks: int
    conversions: Optional[Decimal] = None
    roas: Optional[Decimal] = None
    currency_code: str


class GoogleAdsSummaryResponse(BaseModel):
    """Aggregated ads metrics across all campaigns for a period."""

    period_start: str
    period_end: str
    total_spend: Decimal
    total_impressions: int
    total_clicks: int
    total_conversions: Decimal
    average_roas: Optional[Decimal] = None
    campaigns: list[GoogleAdsCampaignResponse] = []
    daily_metrics: list[GoogleAdsDailyMetricResponse] = []
